src/internal_effect_untraced/dataSource.ts

```diff
@@ -283,7 +283,7 @@ export const fromFunctionBatchedEffect = Debug.untracedMethod((restore) =>
     f: (chunk: Chunk.Chunk<A>) => Effect.Effect<R, Request.Request.Error<A>, Chunk.Chunk<Request.Request.Success<A>>>
   ): DataSource.DataSource<R, A> =>
     makeBatched(name, (requests) =>
-      Effect.serviceWith(completedRequestMap.Tag, (map) =>
+      Effect.flatMap(completedRequestMap.Tag, (map) =>
```

Comment:
I believe there was a bug here with the usage of `Effect.serviceWith` (now Effect.map) which should have previously been `Effect.serviceWithEffect`(now Effect.flatMap) as the "inner" Effect would never have been run. I don't think it's currently covered by the test suite